server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -656,7 +657,10 @@ private FlowInstanceDetailResp cancel(@NotNull FlowInstance flowInstance, Boolea
         return FlowInstanceDetailResp.withIdAndType(id, taskTypeHolder.getValue());
     }
 
-    public FlowInstanceDetailResp approve(@NotNull Long id, String message, Boolean skipAuth) throws IOException {
+    @Transactional(rollbackFor = Exception.class)
```

Comment:
have you tested for the transaction? the flow will be okay when transaction is rollbacked?

Reply:
It works